tests/test_appliancesmanager.py

```diff
@@ -1,123 +1,101 @@
-        http_client_mock,
-        get_appliances_idx + 1,
-        BACKEND_SELECTOR_MOCK.shared_appliances_url,
-        {"WP-CLIENT-BRAND": "DUMMY_BRAND"},
-    )
+async def test_get_shared_appliances_call_contains_wp_client_brand_header(
+    backend_selector_mock: BackendSelector,
+    appliances_manager: AppliancesManager,
+    aioresponses_mock,
+):
+    await appliances_manager.fetch_appliances()
 
-    # ensure that the washer_dryers list is populated
-    assert len(am.washer_dryers) == 2
+    headers = appliances_manager._create_headers(include_wp_brand_name=True)
 
-    await http_client_mock.close_session()
+    aioresponses_mock.assert_called_with(
+        backend_selector_mock.shared_appliances_url, "GET", headers=headers
+    )
 
+    assert headers["WP-CLIENT-BRAND"] == backend_selector_mock.brand.name
 
-@pytest.mark.parametrize(
-    ["owned_response", "shared_response"],
-    [(True, True), (True, False), (False, True), (False, False)],
-)
-async def test_fetch_appliances_returns_true_if_either_method_returns_true(
-    owned_response: bool,
-    shared_response: bool,
-    http_client_mock: AiohttpClientMocker,
+
+async def test_fetch_appliances_returns_owned_and_shared_appliances(
+    appliances_manager: AppliancesManager,
 ):
-    mock_appliancesmanager_get_account_id_get(http_client_mock, BACKEND_SELECTOR_MOCK)
-    mock_appliancesmanager_get_owned_appliances_get(
-        http_client_mock, BACKEND_SELECTOR_MOCK, "12345"
-    )
-    mock_appliancesmanager_get_shared_appliances_get(
-        http_client_mock, BACKEND_SELECTOR_MOCK
-    )
+    await appliances_manager.fetch_appliances()
+
+    # the test data has one owned appliance and one shared appliance
+    # so if this is 2 then we have both
+    assert len(appliances_manager.washer_dryers) == 2
 
-    http_client_mock.create_session(asyncio.get_event_loop())
-    auth = Auth(BACKEND_SELECTOR_MOCK, "email", "secretpass", http_client_mock.session)
 
-    am = AppliancesManager(BACKEND_SELECTOR_MOCK, auth, http_client_mock.session)
-    am._get_shared_appliances = get_mock_coro(shared_response)
-    am._get_owned_appliances = get_mock_coro(owned_response)
+async def test_fetch_appliances_calls_owned_and_shared_methods(
+    appliances_manager: AppliancesManager,
+    backend_selector_mock: BackendSelector,
+    aioresponses_mock,
+):
+    owned_headers = appliances_manager._create_headers()
+    shared_headers = appliances_manager._create_headers(include_wp_brand_name=True)
+
+    await appliances_manager.fetch_appliances()
 
-    result = await am.fetch_appliances()
+    aioresponses_mock.assert_called_with(
+        backend_selector_mock.get_owned_appliances_url(ACCOUNT_ID),
+        "GET",
+        headers=owned_headers,
+    )
 
-    assert result == bool(owned_response or shared_response)
-    await http_client_mock.close_session()
+    aioresponses_mock.assert_called_with(
+        backend_selector_mock.shared_appliances_url, "GET", headers=shared_headers
+    )
```

Comment:
can we also reset `aioresponses_mock`, call `fetch_appliances` again and assert that the call count to `aioresponses_mock` is zero?

Reply:
I didn't follow this comment, why would we do this? fetch_appliances will be calling the endpoints every time, it doesn't cache return appliances

Reply:
Well, I don't remember why I suggested that. Now it does not seem to make sense at all :rofl: